app/assets/javascripts/Components/Modals/submission_file_upload_modal.jsx

```diff
@@ -16,18 +17,69 @@ class SubmissionFileUploadModal extends React.Component {
 
   onSubmit = event => {
     event.preventDefault();
-    this.props.onSubmit(this.state.newFiles, this.state.unzip);
+    if (this.state.newFiles.length == 1) {
+      this.props.onSubmit(this.state.newFiles, this.state.unzip, this.state.single_file_name);
+    } else {
+      this.props.onSubmit(this.state.newFiles, this.state.unzip);
+    }
   };
 
   handleFileUpload = event => {
-    this.setState({newFiles: event.target.files});
+    this.setState({newFiles: event.target.files, single_file_name: event.target.files[0].name});
   };
 
   toggleUnzip = () => {
     this.setState({unzip: !this.state.unzip});
   };
 
+  handleNameChange = event => {
+    this.setState({single_file_name: event.target.value});
+  };
+
   render() {
+    let fileInput;
```

Comment:
Pull this code out into a helper function. Also `fileInput` isn't a good name, since it sounds like the input where users select a file. Something like `fileRenameInput` is better.